### ISSUE TITLE
future times and bad metrics

### ISSUE THREAD

**Issue by grimm26** [livingsocial/rearview#24]
In writing my first monitors, I am not getting any data.  Upon running tshark to watch the outgoing queries to graphite I saw this:

```
330.465848    10.0.2.15 -> 10.11.12.28  HTTP GET /render?from=22:40_20140128&until=22:41_20140128&format=raw&target=alias(stats.gauges.pg.9_1.ca-blah.database.blah_prod_ca.active_sessions,%22active_sessions%22) HTTP/1.1
```

The target took my literal input into the target field in the web ui.  Also, the timestamps provided seem to be in UTC, but I'm in UTC-6, rendering them futuristic, even though is specified the bounds as 1min and now.

**Comment by grimm26** [comment]
In ./lib/rearview/monitor_runner.rb create_from_to_dates() when the default 'now' is supplied as the to_date, it creates a gmt timestamp (Time.now.gmtime) but if I specify a timestamp, it does not convert to gmtime.  It should either ask me what timezone I want or assume localtime.

**Comment by steveakers** [comment]
Rearview assumes everything is in UTC. For our implementation the server times are UTC, graphite times are UTC, and therefore Rearview times are UTC. It is on our list to provide the ability to specify timezone, but have never really had the need for it ourselves.

**Comment by grimm26** [comment]
ok, the timezone thing is on the back burner for you guys.  I may have to fork for the time being, then.  How about the target issue?

**Comment by steveakers** [comment]
The target should be the literal input into the METRICS area of the monitor. Are you sure there is data in graphite for the UTC timestamp you sent for that metric?

**Comment by grimm26** [comment]
https://github.com/livingsocial/rearview/wiki/Overview tells me I can put in alias(foo.bar.baz,"foo_stuff") into the metrics pane.

My graphite data is not in UTC, so when a "now" timestamp is generated (6 hours in the future) it finds no data.  I have to input a timestamp into the until box for "Test Monitor"

**Comment by steveakers** [comment]
Right, so the literal metric alias(foo.bar.baz,"foo_stuff") should be in your target and is not a problem. The reason you aren't getting data is do to the timestamp difference. So you'll need to change the default timezone to match what's store in Graphite. And I misspoke earlier. You can change the timezone in application.rb.

https://github.com/livingsocial/rearview/blob/697592b3d43a4e5a2f3894f0066ff982eb80e9a8/config/application.rb

@talbright Please correct me if I'm wrong here.

**Comment by grimm26** [comment]
I changed the timezone in application.rb and it didn't seem to do anything.

**Comment by talbright** [comment]
I haven't thought through all the issues with supporting multiple timezones. We intentionally store things in UTC though.

If you come up with a good solution feel free to comment on it here and create a PR if it seems like the right approach.

Some things to consider:
- Should it be per user setting or per user?
- What are the ramifications if any for people that log stats to one graphite instance from multiple timezones?
- What if any would the affects be on email alerts? Could the email sent date end up varying from the alert time itself as it appears in the email content/UI? Does that matter?
- How would this affect the cron schedule for the monitor?

And there's probably more I haven't though of yet.

**Comment by tarcieri** [comment]
Ohai! I just got Rearview running, and we're encountering the same problem: our Graphite server is reporting things in Pacific time, and Rearview is using UTC.

Some way of specifying the time zone that the Graphite server is configured with would be super handy, IMO. Perhaps you could configure it along with `config.graphite_connection`?

**Comment by talbright** [comment]
Timing here is ironic, this patch was just submitted today:

https://github.com/livingsocial/rearview-engine/pull/8

I think it might address what you are seeing, but I haven't had time to think it through all the way yet or test it out. If you have a chance maybe you could pull that in and verify? I'm gone on vacation for the next week so I won't be able to do much of anything until I get back.

**Comment by tarcieri** [comment]
Can definitely take a look next week
